system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs

```diff
@@ -1889,6 +1961,173 @@ impl_runtime_apis! {
 		}
 	}
 
+	impl pallet_revive::ReviveApi<Block, AccountId, Balance, Nonce, BlockNumber> for Runtime
+	{
+		fn balance(address: H160) -> U256 {
+			Revive::evm_balance(&address)
+		}
+
+		fn block_gas_limit() -> U256 {
+			Revive::evm_block_gas_limit()
+		}
+
+		fn gas_price() -> U256 {
+			Revive::evm_gas_price()
+		}
+
+		fn nonce(address: H160) -> Nonce {
+			let account = <Runtime as pallet_revive::Config>::AddressMapper::to_account_id(&address);
+			System::account_nonce(account)
```

Comment:
Side note for AHM: When a para sovereign account uses this and we translate it from its child to its sibling representation then they will probably lose access to this derived Eth Address.

@seadanda another reason to hand out free any-proxies.